packages/react-components/src/components/video-player/videoPlayer.tsx

```diff
@@ -49,69 +48,73 @@ export const VideoPlayer = (props: IVideoPlayerProps) => {
   const [timerangesWithSource, setTimerangesWithSource] = useState<VideoTimeRangesWithSource>([]);
   const [timerangesForVideoOnEdgeRaw, setTimerangesForVideoOnEdgeRaw] = useState<VideoTimeRanges>([]);
   const [timerangesForVideoOnEdge, setTimerangesForVideoOnEdge] = useState<VideoTimeRanges>([]);
-  const videoPlayerId = 'iot-' + uuid();
+  const videoPlayerIdRef = useRef<string>('iot-' + uuid());
   const initialPlaybackMode = 'start' in viewport && 'end' in viewport ? PLAYBACKMODE_ON_DEMAND : PLAYBACKMODE_LIVE;
-  const [playbackMode, setPlaybackMode] = useState<typeof PLAYBACKMODE_ON_DEMAND | typeof PLAYBACKMODE_LIVE>(
-    initialPlaybackMode
-  );
   const playbackModeRef = useRef({});
-  playbackModeRef.current = playbackMode;
-  const [videoPlayer, setVideoPlayer] = useState<ReturnType<typeof videojs>>();
-  const [videoErrorDialog, setVideoErrorDialog] = useState<ReturnType<ReturnType<typeof videojs>['createModal']>>();
+  playbackModeRef.current = initialPlaybackMode;
+  const videoPlayerRef = useRef<ReturnType<typeof videojs>>();
+  const videoErrorDialogRef = useRef<ReturnType<ReturnType<typeof videojs>['createModal']>>();
   // Boolean flag to keep track if video is seeked by the user explicitly
   const [isVideoSeeking, setIsVideoSeeking] = useState<boolean>(false);
   let triggerLiveVideoRequesttimeout: ReturnType<typeof setTimeout> | undefined;
   const uploadLiveVideoTimer = 120000; // 2 minutes timer in milli seconds to trigger live video upload every 2 minutes
   const [currentOnDemandSource, setCurrentOnDemandSource] = useState<currentOnDemandSource>();
   let waitForLiveTimeout: ReturnType<typeof setTimeout> | undefined;
-  const [progressBar, setProgressBar] = useState<HTMLDivElement>();
-  const [seekBar, setSeekBar] = useState<HTMLElement>();
-  const [currentTimeIndicator, setCurrentTimeIndicator] = useState<HTMLElement>();
+  const progressBarRef = useRef<HTMLDivElement>();
+  const seekBarRef = useRef<HTMLElement>();
+  const currentTimeIndicatorRef = useRef<HTMLElement>();
 
   useEffect(() => {
     // Fired on video player mount
-    setVideoPlayerStartAndEndTime(viewport, initialPlaybackMode);
-    setPlayer();
+    if (!videoPlayerRef.current) {
+      setVideoPlayerStartAndEndTime(viewport, initialPlaybackMode);
+      setPlayer();
+      updateVideoSource();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!videoPlayerRef.current) return;
     updateVideoSource();
     return () => {
       // Fired on video player unmount
-      if (videoPlayer) {
-        videoPlayer.dispose();
-      }
-      if (triggerLiveVideoRequesttimeout != null) {
-        clearTimeout(triggerLiveVideoRequesttimeout);
-      }
-      if (waitForLiveTimeout != null) {
-        clearTimeout(waitForLiveTimeout);
+      if (videoPlayerRef.current) {
+        videoPlayerRef.current.dispose();
+        videoPlayerRef.current = undefined;
       }
+      clearAllTimeouts();
     };
-  }, []);
+  }, [videoPlayerRef]);
```

Comment:
I believe use ref as dependency won't trigger useEffect rerender

Reply:
Done - updated the PR